File: db4on/test/src/com/db4o/test/types/Callbacks.cs
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

using System;
using j4o.lang;
using com.db4o;
namespace com.db4o.test.types {

   public class Callbacks: RTest {
      
      public Callbacks() : base() {
      }
      public String name;
      
      public void ObjectOnActivate(ObjectContainer container) {
         Console.WriteLine("onActivate");
      }
      
      public void ObjectOnDeactivate(ObjectContainer container) {
         Console.WriteLine("onDeactivate");
      }
      
      public void ObjectOnDelete(ObjectContainer container) {
         Console.WriteLine("onDelete");
      }
      
      public void ObjectOnNew(ObjectContainer container) {
         Console.WriteLine("onNew");
      }
      
      public void ObjectOnUpdate(ObjectContainer container) {
         Console.WriteLine("onUpdate");
      }
      
      public override void Set(int ver) {
         if (ver == 1) {
            name = "OneONEOneONEOneONEOneONEOneONEOneONE";
         } else {
            name = "TwoTWOTwoTWOTwoTWOTwoTWOTwoTWO";
         }
      }
   }
}

File: db4oj/core/src/com/db4o/internal/callbacks/Callbacks.java
/* Copyright (C) 2006   db4objects Inc.   http://www.db4o.com */

package com.db4o.internal.callbacks;

import com.db4o.*;
import com.db4o.internal.*;
import com.db4o.query.*;

public interface Callbacks {

	boolean objectCanNew(Transaction transaction, Object obj);
	boolean objectCanActivate(Transaction transaction, Object obj);
	boolean objectCanUpdate(Transaction transaction, Object obj);
	boolean objectCanDelete(Transaction transaction, Object obj);
	boolean objectCanDeactivate(Transaction transaction, Object obj);

	void objectOnActivate(Transaction transaction, Object obj);
	void objectOnNew(Transaction transaction, Object obj);
	void objectOnUpdate(Transaction transaction, Object obj);
	void objectOnDelete(Transaction transaction, Object obj);
	void objectOnDeactivate(Transaction transaction, Object obj);
	void objectOnInstantiate(Transaction transaction, Object obj);

	void queryOnStarted(Transaction transaction, Query query);
	void queryOnFinished(Transaction transaction, Query query);
	
	boolean caresAboutCommitting();
	boolean caresAboutCommitted();
	
	void classOnRegistered(ClassMetadata clazz);
	
	void commitOnStarted(Transaction transaction, CallbackObjectInfoCollections objectInfoCollections);
	void commitOnCompleted(Transaction transaction, CallbackObjectInfoCollections objectInfoCollections);

    boolean caresAboutDeleting();
    boolean caresAboutDeleted();
    
    void closeOnStarted(ObjectContainer container);
}
